packages/theming/src/utils/getColor.ts

```diff
@@ -0,0 +1,225 @@
+  }
+
+  return color;
+};
+
+/* convert the optional shade + offset to a shade for the given scheme */
+const toShade = (shade?: number | string, offset?: number, scheme?: 'dark' | 'light') => {
+  let _shade;
+
+  if (shade === undefined) {
+    _shade = scheme === 'dark' ? 500 : 700;
+  } else {
+    _shade = parseInt(shade.toString(), 10);
+
+    if (isNaN(_shade)) {
+      throw new TypeError(`Error: unexpected '${typeof shade}' type for color shade "${shade}"`);
+    }
+  }
+
+  return _shade + (offset || 0);
+};
+
+/* convert the given hue object to a hex color */
+const toHex = (
+  hue: Record<string | number, string>,
+  shade?: number | string,
+  offset?: number,
+  scheme?: 'dark' | 'light'
+) => {
+  const _shade = toShade(shade, offset, scheme);
+  let retVal = hue[_shade];
+
+  if (!retVal) {
+    const closestShade = Object.keys(hue)
+      .map(hueShade => parseInt(hueShade, 10))
+      .reduce((previous, current) => {
+        // Find the closest available shade within the given hue
+        return Math.abs(current - _shade) < Math.abs(previous - _shade) ? current : previous;
+      });
+
+    retVal = adjust(hue[closestShade], _shade, closestShade);
+  }
+
+  return retVal;
+};
+
+/* convert the given hue + shade to a color */
+const toColor = (
+  colors: Omit<IGardenTheme['colors'], 'base' | 'variables'>,
+  palette: IGardenTheme['palette'],
+  scheme: 'dark' | 'light',
+  hue: string,
+  shade?: number | string,
+  offset?: number,
+  transparency?: number
+) => {
+  let retVal;
+  let _hue: Hue =
+    colors[hue as keyof typeof colors] /* ex. `hue` = 'primaryHue' */ ||
+    hue; /* ex. `hue` = '#fd5a1e' */
+
+  if (Object.hasOwn(palette, _hue)) {
+    _hue = palette[_hue]; /* ex. `hue` = 'grey' */
+  }
+
+  if (typeof _hue === 'object') {
+    retVal = toHex(_hue, shade, offset, scheme);
+  } else if (valid(_hue)) {
+    if (shade === undefined) {
+      retVal = _hue;
+    } else {
+      const _colors = scale([PALETTE.white, _hue, PALETTE.black])
+        .correctLightness()
+        .colors(PALETTE_SIZE + 2);
+
+      _hue = _colors.reduce<Record<number, string>>((_retVal, color, index) => {
+        if (index > 0 && index <= PALETTE_SIZE) {
+          _retVal[index * 100] = color;
+        }
+
+        return _retVal;
+      }, {});
+
+      retVal = toHex(_hue, shade, offset, scheme);
+    }
+  }
+
+  if (retVal && transparency) {
+    retVal = rgba(retVal, transparency);
+  }
+
+  return retVal;
+};
+
+/* convert the given object + path to a string value */
+const toProperty = (object: object, path: string) => {
+  const retVal = get(object, path);
+
+  if (typeof retVal === 'string') {
+    return retVal;
+  } else if (retVal === undefined) {
+    throw new ReferenceError(`Error: color variable "${path}" is not defined`);
+  } else {
+    throw new TypeError(`Error: unexpected '${typeof retVal}' type for color variable "${path}"`);
+  }
+};
+
+type ColorParameters = {
```

Comment:
💡 Wondering if there is consumer value in externalizing this type.

Reply:
> 💡 Wondering if there is consumer value in externalizing this type.

Yeah, I was wondering the same thing. Let's hold off for now, as we don't export types for the other utilities. If a consumer wants it, we can release the exported type.